generators/cleanup.js

```diff
@@ -159,10 +159,5 @@ function cleanupOldServerFiles(generator, javaDir, testDir, mainResourceDir, tes
     if (generator.isJhipsterVersionLessThan('5.2.2')) {
         generator.removeFile(`${javaDir}config/ElasticsearchConfiguration.java`);
         generator.removeFile('gradle/liquibase.gradle');
-
-        if (generator.authenticationType === 'oauth2' && generator.applicationType === 'microservice') {
```

Comment:
@rainerh It was removed on purpose in the previous commit - check this [conversation](https://github.com/jhipster/generator-jhipster/pull/8176#issuecomment-417708741). 
@mraible / @vishal423 can you please take a look at this?

Reply:
@sendilkumarn Have seen that, but with the current setup the keycloak.yml is referenced in the generated app.yml for microservices with oauth2.

Personally, I'd prefer to be able to run the entire stack for a microservice under development with all requirements, therefore I add most of the times my gateway as well. So I've the need for keycloak with its realm config.
Even if I just need to access the jhipster-registry, which of course has a UI, I'd need keycloak up and running.

If keycloak should not be added at all for microservices, then a other fix is required:

`
diff --git i/generators/server/templates/src/main/docker/app.yml.ejs w/generators/server/templates/src/main/docker/app.yml.ejs
index 508d1e889..26e70ed92 100644
--- i/generators/server/templates/src/main/docker/app.yml.ejs
+++ w/generators/server/templates/src/main/docker/app.yml.ejs
@@ -206,7 +206,7 @@ services:
         # Also set SPRING_CLOUD_CONSUL_CONFIG_FORMAT=files on your apps
         #    - CONFIG_MODE=git
     <%_ } _%>
-    <%_ if (authenticationType === 'oauth2') { _%>
+    <%_ if (authenticationType === 'oauth2' && applicationType !== 'microservice') { _%>
     keycloak:
         extends:
             file: keycloak.yml
`